doc/source/tutorial/optimize.rst

```diff
@@ -34,7 +34,11 @@ The minimum value of this function is 0 which is achieved when
 Note that the Rosenbrock function and its derivatives are included in
 `scipy.optimize`. The implementations shown in the following sections
 provide examples of how to define an objective function as well as its
-jacobian and hessian functions.
+jacobian and hessian functions. Objective functions in `scipy.optimize` 
+expect a numpy array as their first parameter which is to be optimized 
+and must return a float value. The exact calling signature must be
+``f(x, *args)`` where ``x`` represents a numpy array and ``args`` 
+a tuple of additional arguments supplied to the objective function.
```

Comment:
This paragraph was not in great shape to begin with because: the second sentence was important because it serves to introduce the rest of the `minimize` tutorial, the first was unrelated to the second and not very important / parenthetical.

Adding these to the end of the paragraph compounds the problem, so I'd suggest:

- Moving the "Note that the Rosenbrock function and its derivatives are included in `scipy.optimize`." somewhere else - maybe to the first example, where you explicitly define `def rosen`. And perhaps make it a [`note`](https://numpydoc.readthedocs.io/en/latest/format.html#other-points-to-keep-in-mind), explaining that even though they are built-into SciPy, you are writing the function to illustrate how it needs to be structured.
- Move your sentences to the top of the paragraph, and finish it with " The implementations shown in the following sections..."